specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/preview/2023-04-01-preview/migrate.json

```diff
@@ -0,0 +1,15310 @@
+        }
+      ]
+    },
+    "AKSAssessmentSettings": {
+      "type": "object",
+      "description": "Data model of AKS Assessment Settings.",
+      "properties": {
+        "category": {
+          "$ref": "#/definitions/AzureVmCategory",
+          "description": "Gets or sets azure VM category."
+        },
+        "savingsOptions": {
+          "$ref": "#/definitions/SavingsOptions",
+          "description": "Gets or sets savings options."
+        },
+        "consolidation": {
+          "$ref": "#/definitions/ConsolidationType",
+          "description": "Gets or sets consolidation type."
+        },
+        "pricingTier": {
+          "$ref": "#/definitions/PricingTier",
+          "description": "Gets or sets pricing tier."
+        }
+      },
+      "required": [
+        "category",
+        "savingsOptions",
+        "consolidation",
+        "pricingTier"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/AssessmentSettings"
+        }
+      ]
+    },
+    "AKSAssessmentSummary": {
+      "type": "object",
+      "description": "Data model of AKS Assessment Summary.",
+      "properties": {
+        "skuSummary": {
+          "type": "object",
+          "description": "Gets or sets SKU summary.",
+          "additionalProperties": {
+            "$ref": "#/definitions/SummaryDetails"
+          },
+          "readOnly": true
+        },
+        "webAppSummary": {
+          "type": "object",
+          "description": "Gets or sets web app summary.",
+          "additionalProperties": {
+            "$ref": "#/definitions/SummaryDetails"
+          },
+          "readOnly": true
+        },
+        "webServerSummary": {
+          "type": "object",
+          "description": "Gets or sets web server summary.",
+          "additionalProperties": {
+            "format": "int32",
+            "type": "integer"
+          },
+          "readOnly": true
+        },
+        "webAppCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the number of web apps.",
+          "readOnly": true
+        },
+        "webServerCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the number of web servers.",
+          "readOnly": true
+        },
+        "machineCount": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the number of machines.",
+          "readOnly": true
+        },
+        "totalMonthlyCost": {
+          "type": "number",
+          "format": "float",
+          "description": "Gets or sets the total monthly cost.",
+          "readOnly": true
+        }
+      },
+      "required": [
+        "skuSummary",
+        "webAppSummary",
+        "webServerSummary",
+        "webAppCount",
+        "webServerCount",
+        "machineCount",
+        "totalMonthlyCost"
+      ]
```

Comment:
No need to have "required" section since all are read only properties that will be set by the service.

Reply:
agreed, they all are read only properties, but in above context, they are required as service will not skip any of these properties in the response.
so, doesn't it make sense to have them required if we know service wouldn't skip any of them in the response?